api/blocklistener.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+// blockListener defines the block listener in subscribed through API
+type blockListener struct {
+	pendingBlks chan *block.Block
+	cancelChan  chan interface{}
+	streamList  []iotexapi.APIService_StreamBlocksServer
```

Comment:
Consider the problem of concurrent access to the streamList.

Can we use sync.Map instead? Not sure if there's sync.List